torch/csrc/autograd/function.h

```diff
@@ -387,6 +394,37 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
     });
   }
 
+  /// Same as the above `should_compute_output` function but will also
+  /// check whether this edge is needed within the current graph task.
+  bool task_should_compute_output(size_t output_edge_index) const {
+    TORCH_CHECK(output_edge_index < num_outputs(), "Index out of range");
+    const auto& next = next_edges_[output_edge_index];
+    if (next.is_valid()) {
+      const auto exec_info = get_current_graph_task_exec_info();
+      if (exec_info && !exec_info->empty()) {
```

Comment:
exec_info is empty for calls to `.backward()` no?

Reply:
Yup, so we should always return True for that case for when it is empty

Reply:
Which is done just below?

Reply:
If it it is empty, it will just go to the next `return true;`

Reply:
`TORCH_INTERNAL_ASSERT(exec_info);` added

Reply:
we can probably remove this TORCH_INTERNAL_ASSERT below, because it is inside the `if`. My original comment was about adding the assert outside of the `if`, but seems like from the other thread that we won't need that